src/cpp/statistics/rtps/GuidUtils.hpp

```diff
@@ -29,6 +29,17 @@ namespace eprosima {
 namespace fastdds {
 namespace statistics {
 
+/**
+ * Checks whether an entity id corresponds to a builtin monitor service writer.
+ * @param [in] entity_id The entity id to check.
+ * @return true when the entity id corresponds to a builtin monitor service writer.
+ */
+inline bool is_monitor_service_builtin(
+        const fastdds::rtps::EntityId_t& entity_id)
+{
+    return ENTITYID_MONITOR_SERVICE_WRITER == entity_id.to_uint32();
```

Comment:
```suggestion
#ifdef FASTDDS_STATISTICS
    return ENTITYID_MONITOR_SERVICE_WRITER == entity_id.to_uint32();
#else
    (void)entity_id;
    return false;
#endif // ifdef FASTDDS_STATISTICS
```
We should apply something like this or similar to avoid building errors when compiling without the STATISTICS module